ZConfig/datatypes.py

```diff
@@ -277,13 +280,10 @@ def _parse_address(self, s):
 
 
 def float_conversion(v):
-    try:
-        is_str = isinstance(v, basestring)
-    except NameError:
-        # Python 3 support.
-        is_str = isinstance(v, str)
+    # float() will accept both bytes and unicode on both 2 and 3
+    is_str = isinstance(v, text_type) or isinstance(v, binary_type)
     if is_str:
-        if v.lower() in ["inf", "-inf", "nan"]:
+        if v.lower() in (u"inf", u"-inf", u"nan", b'inf', b'-inf', b'nan'):
             raise ValueError(repr(v) + " is not a portable float representation")
```

Comment:
The "depending on the underlying C library" bit remains from the earlier text, which warned that the specific inf/nan spellings did depend on the C library: http://svn.python.org/view/python/trunk/Doc/library/functions.rst?r1=59558&r2=59557&pathrev=59558.  I wonder why it wasn't removed.

The "Float accepts the strings nan, inf and -inf for NaN and positive or negative infinity.  The case and a leading + are ignored as well as a leading - is ignored for NaN." part is about the parsing.  There are [stdlib tests](https://github.com/python/cpython/blob/2.6/Lib/test/test_float.py#L268-L309) for this too.

The "Float always represents NaN and infinity as nan, inf or -inf." part is about the `repr()`.